### PR TITLE
Project Beats: port and add new Sequencer tests

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Adds unit testing coverage for Simple2Sequencer. I ported the existing ProgramSequencerTest and RandomSkipManagerTests as well as added new tests to cover the various APIs.

## Links

https://codedotorg.atlassian.net/browse/SL-845

## Testing story

Unit tests pass locally.

### CODE REVIEW

File: apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    expect(event.length).to.equal(DEFAULT_PATTERN_LENGTH);
+    expect(event.triggered).to.be.false;
+    expect((event as ChordEvent).value).to.equal(chordValue);
+  });
+
+  it('nests together and sequential events correctly', () => {
+    sequencer.newSequence();
+    sequencer.startFunctionContext('when_run');
+
+    // prettier-ignore
```
Comment (breville) [inline apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts:126]: ultra nit: add 
```
// Preserve indentation
```

File: apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    expect(event.type).to.equal('chord');
+    expect(event.id).to.equal(JSON.stringify(chordValue));
+    expect(event.blockId).to.equal('blockId');
+    expect(event.when).to.equal(1);
+    expect(event.length).to.equal(DEFAULT_PATTERN_LENGTH);
+    expect(event.triggered).to.be.false;
+    expect((event as ChordEvent).value).to.equal(chordValue);
+  });
+
+  it('nests together and sequential events correctly', () => {
```
Comment (breville) [inline apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts:121]: ultra nit: consistency in upper-casing (or not) these names :)

**Review by breville — APPROVED** [review]
Fantastic stuff.  Great to see some familiar tests live on, and a bunch of new testing too.  Thanks for seeing this through.

File: apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts:126]: Done!

File: apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts (hunk 2 above)
Comment (sanchitmalhotra126) [reply apps/test/unit/music/player/sequencer/Simple2SequencerTest.ts:121]: Done!

**Pull request merged**